rust/noosphere-core/src/api/client.rs

```diff
@@ -529,3 +531,22 @@ where
         Ok(push_response)
     }
 }
+
+/// Both `reqwest` and `axum` re-export `StatusCode` from the `http` crate.
+///
+/// We're stuck on reqwest@0.11.20 [1] that uses an older version
+/// of `http::StatusCode`, whereas axum >= 0.7 uses the 1.0 release
+/// of several HTTP libraries (`http`, `http-body`, `hyper`) [2], which
+/// we'd like to use as our canonical representation.
+///
+/// This utility converts between the old `reqwest::StatusCode` to the
+/// >=1.0 implementation. Notably, we do not pull in all of `axum`
+/// into the `noosphere-core` crate, only the common underlying
+/// crate `http@1.0.0` (or greater).
+///
+/// [1] https://github.com/subconsciousnetwork/noosphere/issues/686
+/// [2] https://github.com/tokio-rs/axum/blob/5b6204168a676497d2f4188af603546d9ebfe20a/axum/CHANGELOG.md#070-27-november-2023
```

Comment:
This is fine as long as there are no issues compiling to the `wasm32-unknown-unknown` target

Reply:
Looks good: https://github.com/subconsciousnetwork/noosphere/actions/runs/7563832746/job/20596952526?pr=778, the new libs here are `http` (already used by `reqwest`, just older) and `axum-extra`, both utilities and types, specifically pulling these in to avoid pulling in the non-wasm friendly Axum